ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -370,65 +316,63 @@ cacheDataProducer uri env curMod globalEnv inScopeEnv limports = do
 
       rdrElts = globalRdrEnvElts globalEnv
 
-      foldMapM :: (Foldable f, Monad m, Monoid b) => (a -> m b) -> f a -> m b
-      foldMapM f xs = foldr step return xs mempty where
-        step x r z = f x >>= \y -> r $! z `mappend` y
+      -- construct a map from Parents(type) to their fields
+      fieldMap = Map.fromListWith (++) $ flip mapMaybe rdrElts $ \elt -> do
+        flbl <- greFieldLabel elt
+        par <- greParent_maybe elt
+        Just (par,[flbl])
 
-      getCompls :: [GlobalRdrElt] -> IO ([CompItem],QualCompls)
-      getCompls = foldMapM getComplsForOne
+      getCompls :: [GlobalRdrElt] -> ([CompItem],QualCompls)
+      getCompls = foldMap getComplsForOne
 
-      getComplsForOne :: GlobalRdrElt -> IO ([CompItem],QualCompls)
+      getComplsForOne :: GlobalRdrElt -> ([CompItem],QualCompls)
       getComplsForOne (GRE n par True _) =
-          (, mempty) <$> toCompItem par curMod curModNameText n Nothing
+          (toCompItem par curMod curModNameText n Nothing, mempty)
       getComplsForOne (GRE n par False prov) =
-        flip foldMapM (map is_decl prov) $ \spec -> do
+        flip foldMap (map is_decl prov) $ \spec ->
           let originalImportDecl = do
                 -- we don't want to extend import if it's already in scope
                 guard . null $ lookupGRE_Name inScopeEnv n
                 -- or if it doesn't have a real location
                 loc <- realSpan $ is_dloc spec
                 Map.lookup loc importMap
-          compItem <- toCompItem par curMod (printOutputable $ is_mod spec) n originalImportDecl
-          let unqual
+              compItem = toCompItem par curMod (printOutputable $ is_mod spec) n originalImportDecl
+              unqual
                 | is_qual spec = []
                 | otherwise = compItem
               qual
                 | is_qual spec = Map.singleton asMod compItem
                 | otherwise = Map.fromList [(asMod,compItem),(origMod,compItem)]
               asMod = showModName (is_as spec)
               origMod = showModName (is_mod spec)
-          return (unqual,QualCompls qual)
+          in (unqual,QualCompls qual)
 
-      toCompItem :: Parent -> Module -> T.Text -> Name -> Maybe (LImportDecl GhcPs) -> IO [CompItem]
-      toCompItem par m mn n imp' = do
-        docs <- getDocumentationTryGhc packageState curMod n
+      toCompItem :: Parent -> Module -> T.Text -> Name -> Maybe (LImportDecl GhcPs) -> [CompItem]
+      toCompItem par m mn n imp' =
+        -- docs <- getDocumentationTryGhc packageState curMod n
```

Comment:
commented out code

Reply:
@wz1000 is this the line causing the eager deserialization of interfaces?

Reply:
This along with `lookupName` (the `lookupName` is slightly worse since it forces typechecking the iface which is also done lazily, `getDocumentation` will still load in the interface if it wasn't needed before and deserialize the documentation part).

Reply:
still commented out